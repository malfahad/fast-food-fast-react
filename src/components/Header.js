import React from 'react';

const Header = () => (
  <div className="header">
    <h1 className="heading main-title"> Fast Food Fast</h1>
    <h2 className="heading"> Order your food online </h2>
  </div>
);

export default Header;
