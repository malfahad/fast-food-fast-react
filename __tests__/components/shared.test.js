import { shallow } from 'enzyme';
import React from 'react';

import Header, {
  AppHeader, OrderSummary, MenuForm, CustomerMenuButttons, OrderItem,
} from '../../src/components/Shared';

describe('Header', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('AppHeader', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AppHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('OrderSummary', () => {
  it('should render empty order without crashing', () => {
    const props = {
      orderSummary: [],
      total: 0,
    };
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const props = {
      orderSummary: [
        {
          title: 'Fries',
          count: 0,
          subTotal: 1000,
        },
      ],
      total: 1000,
    };
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('MenuForm', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<MenuForm />);
    expect(wrapper).toMatchSnapshot();
  });
});


describe('OrderItem', () => {
  it('should render without crashing', () => {
    const props = {
      itemList: [],
      status: 'Created',
    };
    const wrapper = shallow(<OrderItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('CustomerMenuButtons', () => {
  it('should render add button without crashing', () => {
    const props = {
      id: 1,
      count: 0,
    };
    const wrapper = shallow(<CustomerMenuButttons {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render default without crashing', () => {
    const props = {
      count: 10,
    };
    const wrapper = shallow(<CustomerMenuButttons {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
