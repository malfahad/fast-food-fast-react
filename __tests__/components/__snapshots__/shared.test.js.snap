// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render without crashing 1`] = `ShallowWrapper {}`;

exports[`CustomerMenuButtons should render add button without crashing 1`] = `ShallowWrapper {}`;

exports[`CustomerMenuButtons should render default without crashing 1`] = `ShallowWrapper {}`;

exports[`Header should render without crashing 1`] = `ShallowWrapper {}`;

exports[`MenuForm should render without crashing 1`] = `ShallowWrapper {}`;

exports[`OrderSummary should render empty order without crashing 1`] = `ShallowWrapper {}`;

exports[`OrderSummary should render without crashing 1`] = `ShallowWrapper {}`;
